### PR TITLE
User report feature added

### PR DESCRIPTION
**Pull request by vedansh**

### CODE REVIEW

File: devsnest/app/models/submission.rb [hunk 1]
```diff
@@ -2,4 +2,14 @@
 
 class Submission < ApplicationRecord
   enum status: %i[done notdone doubt]
+  has_one :content
+
+  def self.report(days, userid)
+    res =  Content.joins(:submission).where(submissions: { status: 1, user_id: userid })
+    .where({ updated_at: (Date.today - days)..(Date.today) })
+    .select(:parent_id).group(:parent_id).count
+  end
```
Comment (vedansh) [inline devsnest/app/models/submission.rb:11]: Also filter content by question

**Comment by vedansh** [comment]
Add a controller that shows group data that you are calculating along with total questions solved.

File: devsnest/app/models/submission.rb (hunk 1 above)
Comment (priyamishra2903) [reply devsnest/app/models/submission.rb]: resolved

File: devsnest/app/controllers/api/v1/submissions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    class SubmissionsController < ApplicationController
+      include JSONAPI::ActsAsResourceController
+
+      def create
+        discord_id = params['data']['attributes']['discord_id']
+        question_unique_id = params['data']['attributes']['question_unique_id']
+        user = User.find_by(discord_id: discord_id)
+        content = Content.find_by(unique_id: question_unique_id)
+
+        return render_error('User or Content not found') if user.nil? || content.nil?
```
Comment (newts7) [inline devsnest/app/controllers/api/v1/submissions_controller.rb:14]: Will the lines below ever work?

**Review by newts7 — CHANGES_REQUESTED** [review]

**Pull request merged**